vendor/github.com/ethereum/go-ethereum/whisper/whisperv5/api.go

```diff
@@ -260,19 +297,91 @@ func (api *PublicWhisperAPI) Post(ctx context.Context, req NewMessage) (bool, er
 	// Set key that is used to sign the message
 	if len(req.Sig) > 0 {
 		if params.Src, err = api.w.GetPrivateKey(req.Sig); err != nil {
+			if api.w.deliveryServer != nil {
+				switch isP2P {
```

Comment:
This code with `isP2P` and `Send{P2P|RPC}State` repeats a few times. Maybe worth move into helper function?

Reply:
Hmm, sure, it will make this more DRY.